### ISSUE TITLE
ComposeModifierWithoutDefault triggers for @Composable Modifier factory functions

### ISSUE THREAD

**Issue by dandc87** [slackhq/compose-lints#271]
My codebase has some Modifier helpers that can't be converted to Modifier.Nodes, and want to use the [Composable factory guidance](https://developer.android.com/jetpack/compose/custom-modifiers#create_a_custom_modifier_using_a_composable_modifier_factory) but Slack lints (latest `1.3.1`) don't make that easy.

Currently I have to write this incantation to suppress (task and in-IDE):

```kotlin
@Composable
// noinspection ComposeModifierWithoutDefault
fun @Suppress("ComposeComposableModifier") Modifier.myModifierFactory(
  @SuppressLint("ComposeModifierWithoutDefault") myParameter: String,
): Modifier { ... }
```

I also get a crash if there are no parameters (just the receiver):

```
Stack:
ArrayIndexOutOfBoundsException:Arrays$ArrayList.get(null:-1)
←ModifierWithoutDefaultDetector.visitComposable(ModifierWithoutDefaultDetector.kt:59)
←ComposableFunctionDetector$createUastHandler$1.visitMethod(ComposableFunctionDetector.kt:33)
...
```

**Comment by svenjacobs** [comment]
I think this relates to #369. Composable Modifier factories are now officially permitted by Google so `ComposeModifierWithoutDefault` needs to be updated and `ComposeComposableModifier` removed.

**Comment by ZacSweers** [comment]
Resolved by #378
